libcaf_net/caf/net/stream_transport.hpp

```diff
@@ -104,18 +104,18 @@ class stream_transport : public stream_transport_base<Application> {
     return true;
   }
 
-  bool handle_write_event(endpoint_manager& parent) override {
-    CAF_LOG_TRACE(CAF_ARG2("handle", this->handle().id));
-    // Try to write leftover data.
-    write_some();
-    // Get new data from parent.
-    // TODO: dont read all messages at once - get one by one.
-    for (auto msg = parent.next_message(); msg != nullptr;
-         msg = parent.next_message()) {
-      this->next_layer_.write_message(*this, std::move(msg));
-    }
-    // Write prepared data.
-    return write_some();
+  bool handle_write_event(endpoint_manager& manager) override {
+    CAF_LOG_TRACE(CAF_ARG2("handle", this->handle().id)
+                  << CAF_ARG2("queue-size", write_queue_.size()));
+    auto get_messages = [&] {
+      for (auto msg = manager.next_message(); msg != nullptr;
+           msg = manager.next_message())
+        this->next_layer_.write_message(*this, std::move(msg));
+      return !write_queue_.empty();
+    };
+    while (write_some() && get_messages())
+      ; // nop
+    return !write_queue_.empty();
```

Comment:
This can fetch more messages then we can currently write. Wouldn't it make more sense to only get one message at a time and stop as soon as we can't write to the socket buffer any longer?

Reply:
That would be a good idea. I tried that aproach already, but currently we are not able to actually get to the point where we have more than one message to handle per event. So I thought we could stick with getting all messages.